app/scripts/metamask-controller.js

```diff
@@ -1507,10 +1506,39 @@ export default class MetamaskController extends EventEmitter {
       signatureController: this.signatureController,
       platform: this.platform,
       extension: this.extension,
-      trackTransactionEvents: handleMMITransactionUpdate.bind(
-        null,
-        transactionMetricsRequest,
-      ),
+      hooks: {
+        getTransactions: this.txController.getTransactions.bind(
+          this.txController,
+        ),
+        updateTransactionHash: this.txController.setTxHash.bind(
+          this.txController,
+        ),
+        trackTransactionEvents: handleMMITransactionUpdate.bind(
+          null,
+          transactionMetricsRequest,
+        ),
+        txStateManager: {
```

Comment:
Maybe we can only pass hooks (callbacks) here and generate `txStateManager` mock in `MMIController` constructor.

Reply:
That was my assumption also since we don't have to provide duplicate arguments here and the `txStateManager` becomes an implementation detail for what the NPM package expects.